django/contrib/contenttypes/fields.py

```diff
@@ -251,7 +278,9 @@ def __get__(self, instance, cls=None):
             ct_match = (
                 ct_id == self.get_content_type(obj=rel_obj, using=instance._state.db).id
             )
-            pk_match = ct_match and rel_obj._meta.pk.to_python(pk_val) == rel_obj.pk
+            pk_match = ct_match and (
+                rel_obj._meta.pk.to_python(pk_val)
+            ) == deserialize_pk(serialize_pk(rel_obj))
```

Comment:
@csirmazbendeguz It seems a bit off that we must go through a serde roudtrip here. do you have more details as to why this is necessary?

Reply:
If I recall correctly, this was to convert the UUID to string. I'll have to sort this PR out once composite pks are merged, no need to review it until then.